src/server/scripts/Outland/CoilfangReservoir/SteamVault/boss_hydromancer_thespia.cpp

```diff
@@ -15,172 +15,123 @@
 #include "steam_vault.h"
 
-enum Yells
+enum ThespiaTexts
 {
     SAY_SUMMON                  = 0,
     SAY_AGGRO                   = 1,
     SAY_SLAY                    = 2,
-    SAY_DEAD                    = 3,
+    SAY_DEATH                   = 3
 };
 
-enum Spells
+enum ThespiaSpells
 {
     SPELL_LIGHTNING_CLOUD       = 25033,
     SPELL_LUNG_BURST            = 31481,
-    SPELL_ENVELOPING_WINDS      = 31718
+    SPELL_ENVELOPING_WINDS      = 31718,
+
+    SPELL_WATER_BOLT_VOLLEY     = 34449
 };
 
-enum Events
+enum ThespiaEvents
 {
     EVENT_LIGHTNING_CLOUD       = 1,
     EVENT_LUNG_BURST,
     EVENT_ENVELOPING_WINDS
 };
 
-class boss_hydromancer_thespia : public CreatureScript
+// 17797 - Hydromancer Thespia
+struct boss_hydromancer_thespia : public BossAI
 {
-    public:
-        boss_hydromancer_thespia() : CreatureScript("boss_hydromancer_thespia") { }
-
-        struct boss_thespiaAI : public BossAI
+    boss_hydromancer_thespia(Creature* creature) : BossAI(creature, DATA_HYDROMANCER_THESPIA) { }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        Talk(SAY_AGGRO);
+        BossAI::JustEngagedWith(who);
+
+        events.ScheduleEvent(EVENT_LIGHTNING_CLOUD, 10s, 15s);
+        events.ScheduleEvent(EVENT_LUNG_BURST, 7s, 12s);
+        events.ScheduleEvent(EVENT_ENVELOPING_WINDS, 10s, 15s);
+    }
+
+    void KilledUnit(Unit* who) override
+    {
+        if (who->GetTypeId() == TYPEID_PLAYER)
+            Talk(SAY_SLAY);
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        Talk(SAY_DEATH);
+        _JustDied();
+    }
+
+    void ExecuteEvent(uint32 eventId) override
+    {
+        switch (eventId)
         {
-            boss_thespiaAI(Creature* creature) : BossAI(creature, DATA_HYDROMANCER_THESPIA) { }
-
-            void Reset() override
-            {
-                _Reset();
-            }
-
-            void JustDied(Unit* /*killer*/) override
-            {
-                Talk(SAY_DEAD);
-                _JustDied();
-            }
-
-            void KilledUnit(Unit* who) override
-            {
-                if (who->GetTypeId() == TYPEID_PLAYER)
-                    Talk(SAY_SLAY);
-            }
-
-            void JustEngagedWith(Unit* who) override
-            {
-                Talk(SAY_AGGRO);
-                BossAI::JustEngagedWith(who);
-
-                events.ScheduleEvent(EVENT_LIGHTNING_CLOUD, 15s);
-                events.ScheduleEvent(EVENT_LUNG_BURST, 7s);
-                events.ScheduleEvent(EVENT_ENVELOPING_WINDS, 9s);
-            }
-
-            void ExecuteEvent(uint32 eventId) override
-            {
-                switch (eventId)
-                {
-                    case EVENT_LIGHTNING_CLOUD:
-                        if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0, 30.0f, true))
-                            DoCast(target, SPELL_LIGHTNING_CLOUD);
-                        // cast twice in Heroic mode
-                        if (IsHeroic())
```

Comment:
intended to remove the heroic only part?

Reply:
Yes, not confirmed